zetaclient/evm/evm_signer_test.go

```diff
@@ -0,0 +1,296 @@
+	logger := common.ClientLogger{}
+	ts := &metrics.TelemetryServer{}
+	return NewEVMSigner(
+		corecommon.BscMainnetChain(),
+		stub.EVMRPCEnabled,
+		stub.NewTSSMainnet(),
+		config.GetConnectorABI(),
+		config.GetERC20CustodyABI(),
+		mpiAddress,
+		erc20CustodyAddress,
+		logger,
+		ts)
+}
+
+func getNewEvmChainClient() (*ChainClient, error) {
+	logger := common.ClientLogger{}
+	ts := &metrics.TelemetryServer{}
+	cfg := config.NewConfig()
+	tss := stub.NewTSSMainnet()
+
+	evmcfg := config.EVMConfig{Chain: corecommon.BscMainnetChain(), Endpoint: "http://localhost:8545"}
+	cfg.EVMChainConfigs[corecommon.BscMainnetChain().ChainId] = &evmcfg
+	coreCTX := corecontext.NewZetaCoreContext(cfg)
+	appCTX := appcontext.NewAppContext(coreCTX, cfg)
+
+	return NewEVMChainClient(appCTX, stub.NewZetaCoreBridge(), tss, "", logger, evmcfg, ts)
+}
+
+func getNewOutTxProcessor() *outtxprocessor.Processor {
+	logger := zerolog.Logger{}
+	return outtxprocessor.NewOutTxProcessorManager(logger)
+}
+
+func getCCTX() (*types.CrossChainTx, error) {
+	var cctx crosschaintypes.CrossChainTx
+	err := testutils.LoadObjectFromJSONFile(&cctx, path.Join("../", testutils.TestDataPathCctx, "cctx_56_68270.json"))
+	return &cctx, err
+}
+
+func TestSigner_TryProcessOutTx(t *testing.T) {
+	evmSigner, err := getNewEvmSigner()
+	require.NoError(t, err)
+	cctx, err := getCCTX()
+	require.NoError(t, err)
+	processorManager := getNewOutTxProcessor()
+	mockChainClient, err := getNewEvmChainClient()
+	require.NoError(t, err)
+
+	evmSigner.TryProcessOutTx(cctx, processorManager, "123", mockChainClient, stub.NewZetaCoreBridge(), 123)
+
+	//Check if cctx was signed and broadcasted
+	list := evmSigner.GetReportedTxList()
+	found := false
+	for range *list {
+		found = true
+	}
+	require.True(t, found)
+}
+
+func TestSigner_SignOutboundTx(t *testing.T) {
+	// Setup evm signer
+	evmSigner, err := getNewEvmSigner()
+	require.NoError(t, err)
+
+	// Setup txData struct
+	txData := BaseTransactionData{}
+	cctx, err := getCCTX()
+	require.NoError(t, err)
+	mockChainClient, err := getNewEvmChainClient()
+	require.NoError(t, err)
+	skip, err := txData.SetTransactionData(cctx, mockChainClient, evmSigner.EvmClient(), zerolog.Logger{})
+	require.False(t, skip)
+	require.NoError(t, err)
+
+	t.Run("SignOutboundTx - should successfully sign", func(t *testing.T) {
+		// Call SignOutboundTx
+		tx, err := evmSigner.SignOutboundTx(&txData)
+		require.NoError(t, err)
+
+		// Verify Signature
+		tss := stub.NewTSSMainnet()
+		_, r, s := tx.RawSignatureValues()
+		signature := append(r.Bytes(), s.Bytes()...)
+		hash := evmSigner.EvmSigner().Hash(tx)
+
+		verified := crypto.VerifySignature(tss.Pubkey(), hash.Bytes(), signature)
+		require.True(t, verified)
+	})
+}
+
+func TestSigner_SignRevertTx(t *testing.T) {
+	// Setup evm signer
+	evmSigner, err := getNewEvmSigner()
+	require.NoError(t, err)
+
+	// Setup txData struct
+	txData := BaseTransactionData{}
+	cctx, err := getCCTX()
```

Comment:
Is this CCTX a real revert outtx in BSC?

Reply:
Its not a revert, just a sample cctx.